### ISSUE TITLE
Import declaration conflict not detected with `import { Foo } from` statement

### ISSUE THREAD

**Issue by SamVerschueren** [palantir/tslint#2921]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: (pick one) CLI

##### foo.ts
```ts
interface Foo {
    [key: string]: any;
}
```

##### main.ts
```ts
import { Foo } from './foo';

const Foo = 'bar';
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"no-shadowed-variable": true
	}
}
```

#### Actual behavior

TSLint doesn't catch that `Foo` is being redefined.

#### Expected behavior

TSLint should fail with the following error

> [tslint] Import declaration conflicts with local declaration of 'Foo'

At first I thought this had something to do with the `no-shadowed-variable` rule, but now I don't think it is the case anymore. Whenever you change `main.ts` to something like this

```js
import * as Foo from './foo';

const Foo = 'foo';
```

It detects the conflict and throws an error.

**Comment by ajafff** [comment]
I added this to my PR #2598

Though you won't get a lint error for your code snippet above, because both declarations are in the same scope. But you get a compile error for that anyway.

But the following will work as expected:
```ts
import { Foo } from './foo';
{
    const Foo = 'bar'; // error: shadowed name 'Foo'
}
```

**Comment by SamVerschueren** [comment]
Shouldn't my code snippet produce the following error then?

> [tslint] Import declaration conflicts with local declaration of 'Foo'

If the following snippet produces that error, my initial one should do that as well as it's incorrect.

```ts
import * as Foo from './foo';

const Foo = 'foo';
```

So it's pretty nice that you covered that case, I think it only solves a part of the issue.

Thanks for looking into it btw :)!

**Comment by ajafff** [comment]
>Shouldn't my code snippet produce the following error then?
>> [tslint] Import declaration conflicts with local declaration of 'Foo'

It's actually `[ts] Import declaration conflicts with local declaration of 'Foo'` and does not come from tslint but the typescript compiler.

`no-shadowed-variable` only checks for **shadowed** variables, that means variables in different scopes. That does not include invalid declaration merging, because it's already handled by the compiler.
If you really want this error coming from tslint, you can use `tslint -p . --type-check` which will also check for compile errors.

**Comment by SamVerschueren** [comment]
So I should rather create an issue in the TypeScript repository concerning the issue that it's not throwing the `[ts] Import declaration conflicts with local declaration of 'Foo'` error in case of the following snippet? Because it has nothing to do with TSLint?

```js
import { Foo } from './foo';

const Foo = 'bar';
```

**Comment by ajafff** [comment]
Yes, you should create an issue in the TypeScript repo if you don't see the expected error message.
Though I cannot reproduce that behavior locally. I get the correct error message for both named imports and namespace imports.

**Comment by SamVerschueren** [comment]
Strange. I used the latest TypeScript but it doesn't give me the error. Took a screen capture. But will open this one in the TypeScript repository and link to this thread. Thanks for the information @ajafff!

![tslint](https://user-images.githubusercontent.com/1913805/27166576-0f0c293a-519c-11e7-9bf0-506f2604e17a.gif)
